django_redis/pool.py

```diff
@@ -163,16 +163,27 @@ def get_connection_pool(self, params):
         # explicitly set service_name and sentinel_manager for the
         # SentinelConnectionPool constructor since will be called by from_url
         cp_params = dict(params)
-        cp_params.update(service_name=url.hostname, sentinel_manager=self._sentinel)
-        pool = super().get_connection_pool(cp_params)
-
         # convert "is_master" to a boolean if set on the URL, otherwise if not
         # provided it defaults to True.
-        is_master = parse_qs(url.query).get("is_master")
+        query_params = parse_qs(url.query)
+        is_master = query_params.get("is_master")
         if is_master:
-            pool.is_master = to_bool(is_master[0])
+            cp_params["is_master"] = to_bool(is_master[0])
+        # then remove the "is_master" query string from the URL
+        # so it doesn't interfere with the SentinelConnectionPool constructor
+        if "is_master" in query_params:
+            del query_params["is_master"]
+        new_query = urlencode(query_params, doseq=True)
```

Comment:
why do you need this if you deleted the key from the original `query_params` variable?

Reply:
If this parameter is not deleted, it cannot be correctly parsed as bool type in redis-py, but as str type. This will cause it to use the master node for reading and writing even if is_master=0 is set.

Reply:
yeah, sure, my bad

Reply:
please merge master into your branch and solve conflicts then

Reply:
I have merged and resolved the conflicts